### DISCUSSION TITLE
Can `osm2pgsql-replication init` be used with an extract from osmium-tags-filter?

### DISCUSSION THREAD

**Discussion by wcedmisten** [osm2pgsql-dev/osm2pgsql#2263] [Q&A] [answered]
Hello! I want to keep an updated database of all restaurants on the planet. My current approach is to use [osmium-tags-filter](https://docs.osmcode.org/osmium/latest/osmium-tags-filter.html) to filter the planet file in order to get a filtered PBF containing just the restaurants. This is significantly faster than filtering the data in osm2pgsql.

I'm wondering if using this extract with osm2pgsql-replication and the [regular planet replication URL](https://planet.osm.org/replication/hour/) will result in unexpected behavior? Does `osm2pgsql-replication` need to initialize with all existing nodes on the planet in order to process updates correctly?

I'm imagining a process like this, but I'd like to know if it will result in the correct data:

* Download the standard planet PBF
* Run osmium tags filter to get just the restaurants
* Use that filtered PBF with the flex output for osm2pgsql that also filters for just restaurants
* Run `osm2pgsl-replication init`
* ...time passes
* Whenever an update is needed, run `osm2pgsql-replication update`

Thanks!

**Comment by joto** [comment, marked as answer]
No. This does not work. For instance a new way might come in with a certain tag that you are interested in but the nodes in that way are not in the update and not in your original data.
